VERSIONING.md

```diff
@@ -0,0 +1,58 @@
+In short, quoting from the official document's summary section:
+
+Given a version number MAJOR.MINOR.PATCH, increment the:
+
+1. MAJOR version when you make incompatible API changes,
+2. MINOR version when you add functionality in a backward-compatible manner, and
+3. PATCH version when you make backward-compatible bug fixes.
+
+
+Public APIs
+===========
+
+We consider the following items as a Public API and each one of the changes made against those items in a backward-incompatible way is considered a "breaking change", so we bump the MAJOR version for the release shipping the change.
+
+
+* Public interfaces, classes, and enums that are defined by source code in this project and are included in a published JAR
+  * NOTE: We will likely separate modules for public-APIs and non-public-APIs for the `processor` package in future
+* Configuration property keys
+* Metric names and tags
+
+Anything that is not in the above list IS NOT considered as a Public API.
+
+We DO NOT consider the following items as public APIs (just examples, not a complete list):
+
+* Logging message formats
+* Java thread names
+* Classes from shadowed and relocated dependency (e.g, `protobuf-java`)
+* Classes and resources that are not included in published JAR (e.g, classes related only for testing, benchmarking, etc.)
+* `public` classes, interfaces, and enums that are under the internal modules
+
+
+Backward-compatible
+===================
+
+Backward compatible in this document refers to the following definition:
+
+* An application referring to any public APIs can still be compiled without any errors
+* An application using any public APIs can still expect similar behavior to the previous version that may differ slightly yet within the level that doesn't affect the application's behavior
+* ID-ish strings/values like property keys and metric names kept exactly the same as the previous version
+
+
+Details of the changes
+======================
+
+You can see the list of changes that has been made by particular version at [Releases page](https://github.com/line/decaton/releases).
+
+
+Guide for developers
+====================
+
+It is a developer's (and reviewer's) responsibility to properly bump the required part of the version number along with the patch for the intended changes.
+Upon submitting a PR, one should include the diff for changing the version according to the content of the patch.
+For example, if the current version is `1.2.0`, if the PR contains a newly added public API that doesn't break backward compatibility, the PR should bump the version to `1.3.0`.
+If there is a conflict with changing the version upon merging PR, the greater version should always be adopted. For example, if the previously merged PR bumped the version to `1.3.0` but the PR now being merged has set to `1.2.1`, then the author of the PR being merged should be rebased onto the latest master to ensure the version is fixed at `1.3.0`.
```

Comment:
So the version of the PR now being merged should be `1.3.1` right?

Reply:
No. 

* The PR that has already merged and bumped the version to `1.3.0` contained new feature
* The PR now being merged contains only a compatible fix

A minor version bump covers and changes that requires patch version to get bumped, and major version works the same for minor and patch. So when the version has already bumped to `1.3.0` which has larger minor version than the last release, any patch level changes does not need to touch patch version.

Reply:
Understood, thanks for your explanation.